pkg/auth/cell.go

```diff
@@ -119,6 +120,7 @@ func registerAuthManager(params authManagerParams) (*AuthManager, error) {
 	})
 
 	jobGroup := params.JobRegistry.NewGroup(
```

Comment:
From a pure usage perspective it would be nicer to get rid of the injecting the additional explicit `cell.Scope` - at least for the "default" case where one don't want to set an explicit scope. This should help in keeping the API smaller and easier to understand.

Would it be possible that the job registry itself would already have the correct scope and provide the possibility to retrieve a scoped job - either by injecting a `job.ScopedRegistry` or providing a function like `NewScopedGroup`?

Reply:
We considered this but its difficult. `cell.Scope` does this trick internally where the object you get always contains the name of the current module and with the logger where we already set fields depending on the module name, but the same doesn't work for `job`. That is because `job` has its own package which also imports `cell`, so when we add the logic for this to the module we get an import loop. So we would have to move job into the `cell` package to make that work. That isn't unthinkable, but decided not to do so now.

Reply:
Unfortunate. Thanks for the info!

I think it's OK as long as we document this well enough. (see my other two comments).

And it might be worth to enrich the comment of the function `NewGroup` that the `cell.Scope` can get retrieved via injection.

Reply:
I do like that `job` is in a separate package, just for naming (e.g. `job.Timer` and so on) so I wouldn't fully move it into the `cell` package. 

We could consider seeing how it'd look like if we'd move the required data type definitions from `job` into `cell` (e.g. `job.Registry` becomes `cell.JobRegistry`, with type alias in `job`), but let's try that out separately.